FlowCal/plot.py

```diff
@@ -379,20 +367,18 @@ def density2d(data,
     data_plot = data[:, channels]
 
     # If bins are not specified, try to get bins from data object
-    if (bins is None and hasattr(data_plot, 'hist_bin_edges')
-            and data_plot.hist_bin_edges(0) is not None
-            and data_plot.hist_bin_edges(1) is not None):
-        # Get bin information
-        bdx = data_plot.hist_bin_edges(0)
-        bdy = data_plot.hist_bin_edges(1)
-        # Get bin scaled indices
-        xdx = np.linspace(0, 1, len(bdx))
-        xsx = np.linspace(0, 1, (len(bdx) - 1)/div + 1)
-        xdy = np.linspace(0, 1, len(bdy))
-        xsy = np.linspace(0, 1, (len(bdy) - 1)/div + 1)
-        # Generate sub-sampled bins
-        bins = np.array([np.interp(xsx, xdx, bdx), 
-                            np.interp(xsy, xdy, bdy)])
+    if (hasattr(data_plot, 'hist_bins')
+            and data_plot.hist_bins(0) is not None
+            and data_plot.hist_bins(1) is not None):
+        # Duplicate bins into a two-element list if necessary
+        if not isinstance(bins, list):
+            bins = [bins, bins]
+        # X Axis
+        if bins[0] is None or isinstance(bins[0], int):
+            bins[0] = data_plot.hist_bins(0, nbins=bins[0], log=xlog)
+        # Y axis
+        if bins[1] is None or isinstance(bins[1], int):
+            bins[1] = data_plot.hist_bins(1, nbins=bins[1], log=ylog)
```

Comment:
> can you add the name of the first argument to hist_bins()? (as as you've done with the other 2 arguments) i forgot what that was and had to go look it up.

this is actually true all over the place here; anytime you're using a `FCSData.fxn(channel=x)`, would help remind me what that's doing if you prepended the `channel=` part in the function call.